### PR TITLE
chore: add release/changes.sh to help automate release CHANGELOG creation

### PR DESCRIPTION
**Pull request by devjgm**
This script generates copy-n-pasteable markdown that can serve as the starting point for a human creating the release notes for our releases. It removes "uninteresting" PRs, and outputs markdown with properly linked section headings and PR numbers. A human will then need to manually tweak the output a bit, but this will helpful be a helpful starting point.

---
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/7102)

### CODE REVIEW

File: release/README.md
Comment (coryan) [inline release/README.md:51]: This always points to `main`, even if you are reading on a branch.  I think you can say `/CHANGELOG.md`, i.e., relative links work for GitHub markdown:

https://github.blog/2013-01-31-relative-links-in-markup-files/

File: release/changes.sh
Comment (coryan) [inline release/changes.sh:91]: There is probably a way to do this with `${foo%%,*}` and `${bar##*,}`. Fine as it is for me.

**Review by coryan — APPROVED** [review]
Some optional stuff below.  Approving to unblock you, but if you want another opinion I would understand, I am not the best shell scripting person around.

File: release/changes.sh
Comment (devjgm) [reply release/changes.sh:24]: Probably. I filed an issue to track it.

File: release/changes.sh
Comment (devjgm) [reply release/changes.sh:45]: Right. It's only _needed_ in the `CHANGELOG.md` file, but I tested it and it appears to be fine in the release notes too.

File: release/changes.sh
Comment (devjgm) [reply release/changes.sh:91]: You're probably right, but I'm actually not seeing a clean way to do it. Since there's an _optional_ third comma-separated field, I _think_ the code is clearer with `cut` than with bash string manipulations.

File: release/README.md
Comment (devjgm) [reply release/README.md]: Changed.

**Pull request merged**
